alluxio/worker_ring.py

```diff
@@ -77,23 +81,29 @@ def from_worker_info(worker_info):
             ) from e
 
     @staticmethod
-    def from_worker_hosts(worker_hosts):
+    def from_worker_hosts(
+        worker_hosts, worker_http_port=DEFAULT_HTTP_SERVER_PORT
+    ):
         worker_addresses = set()
         for host in worker_hosts.split(","):
-            worker_address = WorkerNetAddress(host=host)
+            worker_address = WorkerNetAddress(
+                host=host, http_server_port=worker_http_port
+            )
             worker_addresses.add(worker_address)
         return worker_addresses
 
     def dump_main_info(self):
         return (
-            "WorkerNetAddress{{host={}, containerHost={}, rpcPort={}, dataPort={}, webPort={}, domainSocketPath={}}}"
+            "WorkerNetAddress{{host={}, containerHost={}, rpcPort={}, "
+            "dataPort={}, webPort={}, domainSocketPath={}, httpServerPort={}}}"
         ).format(
             self.host,
             self.container_host,
             self.rpc_port,
             self.data_port,
             self.web_port,
             self.domain_socket_path,
+            self.http_server_port,
```

Comment:
worker_string = worker_address.dump_main_info()
                for i in range(weight):
                    hash_key = self._hash(worker_string, i)
                    hash_ring[hash_key] = worker_address
                    
dump_main_info is used to construct the hash ring and you are changing the hash ring which may make all "loaded files" useless

Reply:
and this suppose to be same as the Alluxio server side which i am not quite sure whether they have changed on Alluxio server side or not

Reply:
I think Lu means we need http port in dump_main_info